### PR TITLE
Re-enable stats for scores and drops

### PR DESCRIPTION
**Pull request by SibiAkkash**

### CODE REVIEW

File: server/tournament/match_stats_min.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+) -> tuple[int, MatchStats | dict[str, str]]:
+    if match_event.block_by_id is None:
+        return 422, {"message": "Block by is needed"}
+
+    try:
+        block_by = Player.objects.get(id=match_event.block_by_id)
+    except Player.DoesNotExist:
+        return 422, {"message": "Invalid Block by"}
+
+    if team.id == match.stats.current_possession.id:
```
Comment (Joe2k) [inline server/tournament/match_stats_min.py:118]: this is wrong? current possession is the team on O?

File: server/tournament/match_stats_min.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    new_match_event = MatchEvent(
+        stats=match.stats,
+        team=team,
+        started_on=MatchEvent.Mode.OFFENSE,  # ignore this for now, not used
+        block_by=block_by,
+        type=MatchEvent.EventType.BLOCK,
+    )
+    new_match_event.save()
+
+    if match.team_1 is not None and match.team_1.id == team.id:
```
Comment (Joe2k) [inline server/tournament/match_stats_min.py:130]: whats the need for if condition? both conditions result in same thing to do?

File: server/tournament/match_stats_min.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    if block_event.team.id == match.team_2.id:
+        match.stats.current_possession = match.team_1
+
+    block_event.delete()
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_undo(match: Match) -> tuple[int, MatchStats | message_response]:
```
Comment (Joe2k) [inline server/tournament/match_stats_min.py:158]: you can put this on top and then followed by handle_score_undo and handle_block_undo so readablity be better

File: server/tournament/match_stats_min.py (hunk 1 above)
Comment (SibiAkkash) [reply server/tournament/match_stats_min.py:118]: so current possession is team on O. `team` here is the team of the player who got the block right ? So the check will fail if the team on O gets a block, which cant happen

File: server/tournament/match_stats_min.py (hunk 2 above)
Comment (SibiAkkash) [reply server/tournament/match_stats_min.py:130]: yeah not needed. I took the code from your `match_stats.py` file, didn't check fully lol

**Review by Joe2k — APPROVED** [review]

**Pull request merged**